src/components/NodeLink/Legend.vue

```diff
@@ -0,0 +1,428 @@
+<script>
+import * as d3 from "d3";
+
+export default {
+  components: {
+  },
+
+  props: {
+    app: {
+      type: Object,
+      required: true
+    },
+    provenance: {
+      type: Object,
+      required: true
+    },
+    graphStructure: {
+      type: Object,
+      default: () => {}
+    },
+    nodeColorScale: {
+      type: Function,
+      default: null
+    },
+    linkColorScale: {
+      type: Function,
+      default: null
+    },
+    glyphColorScale: {
+      type: Function,
+      default: null
+    },
+    linkWidthScale: {
+      type: Function,
+      default: null
+    },
+    multiVariableList: {
+      type: Array,
+      default: () => [],
+    },
+    linkVariableList: {
+      type: Array,
+      default: () => [],
+    },
+    barVariables: {
+      type: Array,
+      default: () => [],
+    },
+    glyphVariables: {
+      type: Array,
+      default: () => [],
+    },
+    widthVariables: {
+      type: Array,
+      default: () => [],
+    },
+    colorVariables: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      svgHeight: 150,
+      yAxisPadding: 10,
+      varPadding: 10,
+    };
+  },
+
+  mounted() {
+    this.setUpPanel()
+  },
+
+  computed: {
+    properties() {
+      const {
+        app,
+        provenance,
+        graphStructure,
+        linkWidthScale,
+        multiVariableList,
+        linkVariableList,
+        barVariables,
+        glyphVariables,
+        widthVariables,
+        colorVariables,
+      } = this;
+      return {
+        app,
+        provenance,
+        graphStructure,
+        linkWidthScale,
+        multiVariableList,
+        linkVariableList,
+        barVariables,
+        glyphVariables,
+        widthVariables,
+        colorVariables,
+      };
+    },
+  },
```

Comment:
Depending on why exactly we're using this mechanism, we may want to just get rid of it. This is error prone, and its use has to be weighed against the benefit of using it.

Reply:
What exactly are the potential bugs that this would cause?

Reply:
If you forget to add something here that you then depend on downstream, or more immediately, if you put something in the top part but not the bottom part.

In general, it's better to just directly express the dependencies in each function rather than go through a mechanism like this. Originally I had something like this to get around some limitation of Vue that I probably shouldn't have been messing with (something like trying to make a side effect reactive by making it depend on a bundle like this, and attaching it to something invisible in the DOM to force updates--it was ugly and wrong).

Reply:
Makes sense. I'll start to break this up as I work through the other issues then so the dependencies for each function are more apparent.